chkcryptoboot_hook

```diff
@@ -19,9 +20,35 @@ check_mbr ()
     fi
 }
 
+check_core ()
+{
+    if [ -n "$BOOTDISK" -a -s /etc/chkcryptoboot/grub-core.sha512 ]; then
+	resolve_device "$BOOTDISK"
+	if [ -n "$BOOT_PARTITION" ]; then
+	    dd if=$BOOT_PARTITION of=/etc/chkcryptoboot/grub-core bs=512 count=62
+	else
+	    dd if=$BOOTDISK of=/etc/chkcryptoboot/grub-core skip=1 bs=512 count=62
+	fi
+	cd /etc/chkcryptoboot
+	sha512sum -c grub-core.sha512
+	if [ $? -ne 0 ]; then
+	    echo -e "\nCHKCRYPTOBOOT ALERT!\n"
+	    echo -e "CHANGES HAVE BEEN DETECTED IN THE BOOT LOADER CORE!\n"
+            cd -
+	    return 1
+	else
+	    echo "Your boot loader core hash was verified successfully."
+            cd -
+	    return 0
+	fi
+    fi
+}
+
 check_efi ()
 {
     if [ -n "$ESP" -a -s /etc/chkcryptoboot/efistub.sha512 ]; then
+	device=`findmnt ${ESP} --fstab -o source -n`
+	resolve_device "${device}"
```

Comment:
We can use resolve_device for the ESP in the context of supporting UUID|PARTLABEL|LABEL|PARTUUID for example. I would be fine with running resolve_device on ESP, but, it's either that or findmnt. Both should not be used at the same time. Also, we should reflect this on the configuration, that the user can either provide /dev/* paths or UUID=<esp fs uuid>, for example.

Reply:
This is the only line that actually has a relevance for #10